pagerduty/resource_pagerduty_event_orchestration_path_router.go

```diff
@@ -0,0 +1,394 @@
+package pagerduty
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/heimweh/go-pagerduty/pagerduty"
+	"log"
+	"time"
+)
+
+func resourcePagerDutyEventOrchestrationPathRouter() *schema.Resource {
+	return &schema.Resource{
+		Read:   resourcePagerDutyEventOrchestrationPathRouterRead,
+		Create: resourcePagerDutyEventOrchestrationPathRouterCreate,
+		Update: resourcePagerDutyEventOrchestrationPathRouterUpdate,
+		Delete: resourcePagerDutyEventOrchestrationPathRouterDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourcePagerDutyEventOrchestrationPathRouterImport,
+		},
+		Schema: map[string]*schema.Schema{
+			"type": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"parent": {
+				Type:     schema.TypeList,
+				Required: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: PagerDutyEventOrchestrationPathParent,
+				},
+			},
+			"sets": {
+				Type:     schema.TypeList,
+				Required: true,
+				MaxItems: 1, // Router can only have 'start' set
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"rules": {
+							Type:     schema.TypeList,
+							Optional: true,
```

Comment:
rules is made optional, because API creates an empty list of rules if it is not passed. We find this will be especially useful when we import a brand new router.